core/trino-main/src/main/java/io/trino/operator/DirectExchangeClientConfig.java

```diff
@@ -35,6 +35,7 @@
     private int clientThreads = 25;
     private int pageBufferClientMaxCallbackThreads = 25;
     private boolean acknowledgePages = true;
+    private DataSize deduplicationBufferSize = DataSize.of(32, Unit.MEGABYTE);
```

Comment:
I am not convinced we need this one. Why can't we have just `maxBufferSize` and set it to different value if we decide to enable QUERY-based retries and to different value without retries?

And even if we want specific property I would not use `deuplication` in it. Feels to low-level.
Maybe `exchange.query-retries-max-buffer-size`?

Reply:
The deduplication buffer is used on coordinator and we may want to configure it differently. For example for pipelined execution we may want to have a higher (or lower) buffer size.

Reply:
Yeah true - both are used in same query - makes sense